frontend/src/scenes/pipeline/AppsManagement.tsx

```diff
@@ -0,0 +1,394 @@
+import { LemonBanner, LemonDivider, LemonLabel, LemonTable, Tooltip } from '@posthog/lemon-ui'
+import { Popconfirm } from 'antd'
+import { useActions, useValues } from 'kea'
+import { IconDelete, IconLock, IconLockOpen } from 'lib/lemon-ui/icons'
+import { LemonButton } from 'lib/lemon-ui/LemonButton'
+import { LemonInput } from 'lib/lemon-ui/LemonInput'
+import { Link } from 'lib/lemon-ui/Link'
+import { preflightLogic } from 'scenes/PreflightCheck/preflightLogic'
+import { SceneExport } from 'scenes/sceneTypes'
+
+import { PluginInstallationType, PluginType } from '~/types'
+
+import { appsManagementLogic } from './appsManagementLogic'
+import { RenderApp } from './utils'
+
+export const scene: SceneExport = {
+    component: AppsManagement,
+    logic: appsManagementLogic,
+}
+
+export function AppsManagement(): JSX.Element {
+    const { canInstallPlugins, canGloballyManagePlugins } = useValues(appsManagementLogic)
+    const { globalPlugins, localPlugins } = useValues(appsManagementLogic)
+
+    if (!canInstallPlugins || !canGloballyManagePlugins) {
+        return <>You don't have permission to manage apps.</>
+    }
+
+    return (
+        <div className="pipeline-apps-management-scene">
+            <OutOfSyncApps />
+            <InstallFromUrl />
+            <InstallLocalApp />
+            <InstallSourceApp />
+
+            <LemonDivider />
+
+            {globalPlugins && (
+                <>
+                    <LemonLabel> Global Apps </LemonLabel>
+                    <p>These apps can be used by any organizations.</p>
+                    <AppsTable plugins={globalPlugins} />
+                </>
+            )}
+
+            {localPlugins && (
+                <>
+                    <LemonLabel> Local Apps </LemonLabel>
+                    <p>These apps can only be used by organizations that have an existing plugin config.</p>
+                    <AppsTable plugins={localPlugins} />
+                </>
+            )}
+        </div>
+    )
+}
+
+type RenderAppsTable = {
+    plugins: PluginType[]
+}
+
+function AppsTable({ plugins }: RenderAppsTable): JSX.Element {
+    const { unusedPlugins } = useValues(appsManagementLogic)
+    const { uninstallPlugin, patchPlugin } = useActions(appsManagementLogic)
+
+    // TODO: row expansion to show the source code and allow updating source apps
```

Comment:
TODO left

Reply:
yup planning to do that in a follow-up PR, this one got big enough